lib/adagrams.rb

```diff
@@ -0,0 +1,141 @@
+require 'csv'
+
+
+def draw_letters
+  hand = Array.new(10)
+
+  letter_pool = [
+      {letter: "A", count: 9},
+      {letter: "B", count: 2},
+      {letter: "C", count: 2},
+      {letter: "D", count: 4},
+      {letter: "E", count: 12},
+      {letter: "F", count: 2},
+      {letter: "G", count: 3},
+      {letter: "H", count: 2},
+      {letter: "I", count: 9},
+      {letter: "J", count: 1},
+      {letter: "K", count: 1},
+      {letter: "L", count: 4},
+      {letter: "M", count: 2},
+      {letter: "N", count: 6},
+      {letter: "O", count: 8},
+      {letter: "P", count: 2},
+      {letter: "Q", count: 1},
+      {letter: "R", count: 6},
+      {letter: "S", count: 4},
+      {letter: "T", count: 6},
+      {letter: "U", count: 4},
+      {letter: "V", count: 2},
+      {letter: "W", count: 2},
+      {letter: "X", count: 1},
+      {letter: "Y", count: 2},
+      {letter: "Z", count: 1}
+  ]
+
+  hand.each_index do |i|
+    letter_index = rand(0...letter_pool.length)
+
+    while letter_pool[letter_index][:count] == 0
+      letter_index = rand(0...letter_pool.length)
+    end
+
+    hand[i] = letter_pool[letter_index][:letter]
+    letter_pool[letter_index][:count] -= 1
+  end
+
+  return hand
+end
+
+def uses_available_letters?(input, letters_in_hand)
+  raise ArgumentError.new("input is nil") if input.nil?
+  raise ArgumentError.new("letters_in_hand is nil") if letters_in_hand.nil?
+
+
+  input_array = input.upcase.split("").sort
+  input_letter_count = {}
+  inter_letter_count = {}
+
+  intersection_with_dupes = (letters_in_hand - (letters_in_hand - (input_array & letters_in_hand)))
+
+  input_array.each do |letter|
+    input_letter_count[letter.to_sym] = input_array.count(letter)
+  end
+
+  intersection_with_dupes.each do |letter|
+    inter_letter_count[letter.to_sym] = intersection_with_dupes.count(letter)
+  end
+
+  uses_available = true
+  if(input_letter_count.length == inter_letter_count.length)
+    input_letter_count.each do |letter, count|
+      uses_available = uses_available && count <= inter_letter_count[letter]
+    end
+  else
+    uses_available = false
+  end
+
+  return uses_available
+end
+
+def score_word(word)
+
+  raise ArgumentError.new("words is nil") if word.nil?
+
+  total_score = 0
+
+  word.upcase.each_char do |letter|
```

Comment:
While this `case` statement works, it means that the information about which letter has which score is locked into this piece of code, and can't easily be used elsewhere. For example, if you wanted to display the value of each letter in a hand, you would need to repeat this work.

An alternative approach would be to store the letter scores in a hash, something like this:

```ruby
LETTER_SCORES = {
  "A" => 1
  "B" => 3,
  "C" => 3,
  "D" => 2,
  # ...
}